src/view.js

```diff
@@ -11,13 +11,180 @@ var isFunction = require("can-util/js/is-function/is-function"),
+};
+
+// ##### checkText
+// Makes sure there's a template, if not, have `steal` provide a warning.
+var checkText = function (text, url) {
+	if (!text.length) {
+
+		// _removed if not used as a steal module_
+
+		//!steal-remove-start
+		can.dev.log("can/view/view.js: There is no template or an empty template at " + url);
+		//!steal-remove-end
+
+		throw new Error("can.view: No template or empty template:" + url);
+	}
+};
+
+// ##### get
+// get a deferred renderer for provided url
+/**
+ * @hide
+ * @function get
+ * @param {String | Object} obj url string or object with url property
+ * @param {Boolean} async If the ajax request should be asynchronous.
+ * @return {can.Deferred} a `view` renderer deferred.
+ */
+var	getRenderer = function (obj, async) {
+	// If `obj` already is a renderer function just resolve a Deferred with it
+	if(isFunction(obj)) {
+		var def = can.Deferred();
+		return def.resolve(obj);
+	}
+
+	var url = typeof obj === 'string' ? obj : obj.url,
+		suffix = (obj.engine && '.' + obj.engine) || url.match(/\.[\w\d]+$/),
+		type,
+	// If we are reading a script element for the content of the template,
+	// `el` will be set to that script element.
+		el,
+	// A unique identifier for the view (used for caching).
+	// This is typically derived from the element id or
+	// the url for the template.
+		id;
+
+	//If the url has a #, we assume we want to use an inline template
+	//from a script element and not current page's HTML
+	if (url.match(/^#/)) {
+		url = url.substr(1);
+	}
+	// If we have an inline template, derive the suffix from the `text/???` part.
+	// This only supports `<script>` tags.
+	if (el = document.getElementById(url)) {
+		suffix = '.' + el.type.match(/\/(x\-)?(.+)/)[2];
+	}
+
+	// If there is no suffix, add one.
+	if (!suffix && !$view.cached[url]) {
+		url += suffix = $view.ext;
+	}
+
+	// if the suffix was derived from the .match() operation, pluck out the first value
+	if (can.isArray(suffix)) {
+		suffix = suffix[0];
+	}
 
+	// Convert to a unique and valid id.
+	id = $view.toId(url);
+
+	// If an absolute path, use `steal`/`require` to get it.
+	// You should only be using `//` if you are using an AMD loader like `steal` or `require` (not almond).
+	if (url.match(/^\/\//)) {
+		url = url.substr(2);
+		url = !window.steal ?
+			url :
+			steal.config()
+				.root.mapJoin("" + steal.id(url));
+	}
+
+	// Localize for `require` (not almond)
+	if (window.require) {
+		if (require.toUrl) {
+			url = require.toUrl(url);
+		}
+	}
+
+	// Set the template engine type.
+	type = $view.types[suffix];
+
+	// If it is cached,
+	if ($view.cached[id]) {
+		// Return the cached deferred renderer.
+		return $view.cached[id];
+
+		// Otherwise if we are getting this from a `<script>` element.
+	} else if (el) {
+		// Resolve immediately with the element's `innerHTML`.
+		return $view.registerView(id, el.innerHTML, type);
+	} else {
+		// Make an ajax request for text.
+		var d = new can.Deferred();
+		jQuery.ajax({
```

Comment:
`can.ajax` support `async: false`?

Reply:
Will need to update https://github.com/canjs/can-ajax both 1.X and 2.X versions with this.